File: etc/recurvie_function/python/hanoi_tower.py
def hanoi(num, n_from, to, other):
    if num == 0: return
    print("num: {}, from: {}, other: {}, to: {}".format(num, n_from, to, other))
    hanoi(num - 1, n_from, other, to)
    hanoi(num - 1, other, to, n_from)


hanoi(4, 0, 1, 2)

File: data_structure&algorithm/etc/recurvie_function/hanoi-tower.js
function hanoi (num, from, to, other) {
    if (num === 0)  return;
    console.log(`num: ${num}, from: ${from}, to: ${to}, other: ${other}.`)
    hanoi (num - 1, from, other, to);
    hanoi (num - 1, other, to, from);
}

hanoi(4, 0, 1, 2);